### ISSUE TITLE
Can we support Rive animations like we do Lottie?

### ISSUE THREAD

**Issue by JonnyBurger** [remotion-dev/remotion#1718]
https://rive.app/

**Comment by kaf-lamed-beyt** [comment]
I'm thinking of something @JonnyBurger 

This would be a component that accepts a `src` prop that'll point to the Rive animation, and it'll integrate Remotion's `useVideo` API in a particular way, yeah? 

If that's the case, i think I can go over some of the APIs in the React runtime of Rive, and spin something up.

What do you think?

**Comment by kaf-lamed-beyt** [comment]
@JonnyBurger, this is me touching base with you, to know if i can go ahead with this.

**Comment by JonnyBurger** [comment]
It seems like we need to use the `.advance()` API to map Remotion's declarative `useCurrentFrame()` to Rive's imperative API: https://help.rive.app/runtimes/overview/web-js/low-level-api-usage#advancing-animations to do this. The Lottie implementation is a good reference for it though.

@kaf-lamed-beyt this is pretty complicated, are you sure you have a plan for it?
I don't think it should use the useVideo() API, actually we don't even have such API.

**Comment by kaf-lamed-beyt** [comment]
> I don't think it should use the useVideo() API, actually we don't even have such API.

Oops! It should be `useVideoConfig`. That was a mistake. 

I think it could be possible, since we can use the `useVideoConfig` API to get the current frame rate and the duration of animation by importing `fps` and `durationInFrames` from it.

We can have a couple of Effects that would take charge of loading the Rive animation, setting up the Rive animation player and plays the animation when the canvas — from Rive — ref variables are set. Ideally, we can have these state variables `animation` and `canvasRef` to keep track of the change, such that the effect only runs when it detects any change in these variables.

Another logic that we may need to consider would be to sync the Rive animation with Remotion's timeliness for a typical video by setting its time frame (that of the Rive animation) with the current frame of the video.

I don't know if this makes any sense 🤔

**Comment by kaf-lamed-beyt** [comment]
Lemme take a look at the `.advance()` API now, though.

**Comment by kaf-lamed-beyt** [comment]
...and yes, I think what I explained is a bit similar to the `advance()` API approach. 

We can tap into the linear time method — `setLinearTime()` — and sync it with `fps` and `durationInFrames` 

Something like this 👇🏼 

```js
    setLinearTime(fps * durationInFrames) / 30
```

**Comment by JonnyBurger** [comment]
Oh got it! That makes total sense.
Where did you see the `setLinearTime()` method? I did not find it!

Anyway, I'm assigning this issue to you! Good luck and let me know if you need help 🙌

**Comment by kaf-lamed-beyt** [comment]
> Where did you see the setLinearTime() method? I did not find it!

My bad. I must have confused it with the Linear animation instance. Speaking about that now, I also noticed that we can use the `advance()` API too. You're right all along.

I'll try spinning up something, and I'll update you as I progress. I'll also run back here to ask for guidance when I encounter any issue. 

Thanks again for this opportunity.

**Comment by JonnyBurger** [comment]
Thank you and good luck! :raised_hands:

**Comment by kaf-lamed-beyt** [comment]
@JonnyBurger,

I'm having an issue with installing the rive dependency, coupled with react, react-dom, and remotion too. But, I keep getting this error in the terminal

![image](https://user-images.githubusercontent.com/48127194/223645592-515c8167-c282-44da-8943-e17075acfc7a.png)

I even tried clearing my cache with 👇🏼 

```bash 
npm cache clean --force 
```

But, the error persisted.

**Comment by JonnyBurger** [comment]
Since we use `pnpm`, npm will not work in our monorepo - can you try `pnpm i rive` instead?

See here for pnpm setup: https://www.remotion.dev/docs/contributing/#setup

**Comment by kaf-lamed-beyt** [comment]
I must have thought mono repo convention used npm because I didn't see any pnpm lock file. But thank you for pointing that out. I'll proceed now. 🙌🏽

**Comment by JonnyBurger** [comment]
Fixed by #1904
